### ISSUE TITLE
IDE Crash

### ISSUE THREAD

**Issue by jsiebern** [reasonml-editor/reasonml-idea-plugin#79]
plugin version: `0.58`

### Description
My IDE just crashed with the following error:

```
com.intellij.psi.impl.source.tree.LeafPsiElement cannot be cast to com.intellij.psi.impl.source.tree.CompositeElement
java.lang.ClassCastException: com.intellij.psi.impl.source.tree.LeafPsiElement cannot be cast to com.intellij.psi.impl.source.tree.CompositeElement
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:160)
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:153)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:75)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:27)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:972)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:954)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:197)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.diff(DiffTree.java:61)
	at com.intellij.psi.impl.BlockSupportImpl.diffTrees(BlockSupportImpl.java:336)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:996)
	at com.intellij.lang.impl.PsiBuilderImpl.b(PsiBuilderImpl.java:910)
	at com.intellij.lang.impl.PsiBuilderImpl.getTreeBuilt(PsiBuilderImpl.java:901)
	at com.reason.lang.CommonParser.parse(CommonParser.java:31)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:88)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:49)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.lambda$ensureParsed$0(LazyParseableElement.java:193)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:544)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:247)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:40)
	at com.intellij.psi.impl.BlockSupportImpl.isReplaceWholeNode(BlockSupportImpl.java:359)
	at com.intellij.psi.impl.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:305)
	at com.intellij.psi.impl.BlockSupportImpl.makeFullParse(BlockSupportImpl.java:241)
	at com.intellij.psi.impl.BlockSupportImpl.reparse(BlockSupportImpl.java:101)
	at com.intellij.psi.impl.DocumentCommitThread.doCommit(DocumentCommitThread.java:682)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$commitUnderProgress$4(DocumentCommitThread.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1134)
	at com.intellij.psi.impl.DocumentCommitThread.commitUnderProgress(DocumentCommitThread.java:473)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$pollQueue$1(DocumentCommitThread.java:285)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.psi.impl.DocumentCommitThread.pollQueue(DocumentCommitThread.java:285)
	at com.intellij.psi.impl.DocumentCommitThread.run(DocumentCommitThread.java:245)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.lang.ClassCastException: com.intellij.psi.impl.source.tree.LeafPsiElement cannot be cast to com.intellij.psi.impl.source.tree.CompositeElement
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:160)
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:153)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:75)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:27)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:972)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:954)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:197)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.diff(DiffTree.java:61)
	at com.intellij.psi.impl.BlockSupportImpl.diffTrees(BlockSupportImpl.java:336)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:996)
	at com.intellij.lang.impl.PsiBuilderImpl.b(PsiBuilderImpl.java:910)
	at com.intellij.lang.impl.PsiBuilderImpl.getTreeBuilt(PsiBuilderImpl.java:901)
	at com.reason.lang.CommonParser.parse(CommonParser.java:31)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:88)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:49)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.lambda$ensureParsed$0(LazyParseableElement.java:193)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:544)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:247)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:40)
	at com.intellij.psi.impl.BlockSupportImpl.isReplaceWholeNode(BlockSupportImpl.java:359)
	at com.intellij.psi.impl.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:305)
	at com.intellij.psi.impl.BlockSupportImpl.makeFullParse(BlockSupportImpl.java:241)
	at com.intellij.psi.impl.BlockSupportImpl.reparse(BlockSupportImpl.java:101)
	at com.intellij.psi.impl.DocumentCommitThread.doCommit(DocumentCommitThread.java:682)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$commitUnderProgress$4(DocumentCommitThread.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1134)
	at com.intellij.psi.impl.DocumentCommitThread.commitUnderProgress(DocumentCommitThread.java:473)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$pollQueue$1(DocumentCommitThread.java:285)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.psi.impl.DocumentCommitThread.pollQueue(DocumentCommitThread.java:285)
	at com.intellij.psi.impl.DocumentCommitThread.run(DocumentCommitThread.java:245)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.lang.ClassCastException: com.intellij.psi.impl.source.tree.LeafPsiElement cannot be cast to com.intellij.psi.impl.source.tree.CompositeElement
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:160)
	at com.intellij.psi.impl.DiffLog$InsertEntry.<init>(DiffLog.java:153)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:75)
	at com.intellij.psi.impl.DiffLog.nodeInserted(DiffLog.java:27)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:972)
	at com.intellij.lang.impl.PsiBuilderImpl$ConvertFromTokensToASTBuilder.nodeInserted(PsiBuilderImpl.java:954)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:197)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.build(DiffTree.java:151)
	at com.intellij.util.diff.DiffTree.diff(DiffTree.java:61)
	at com.intellij.psi.impl.BlockSupportImpl.diffTrees(BlockSupportImpl.java:336)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:996)
	at com.intellij.lang.impl.PsiBuilderImpl.b(PsiBuilderImpl.java:910)
	at com.intellij.lang.impl.PsiBuilderImpl.getTreeBuilt(PsiBuilderImpl.java:901)
	at com.reason.lang.CommonParser.parse(CommonParser.java:31)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:88)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:49)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.lambda$ensureParsed$0(LazyParseableElement.java:193)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:544)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:247)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:40)
	at com.intellij.psi.impl.BlockSupportImpl.isReplaceWholeNode(BlockSupportImpl.java:359)
	at com.intellij.psi.impl.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:305)
	at com.intellij.psi.impl.BlockSupportImpl.makeFullParse(BlockSupportImpl.java:241)
	at com.intellij.psi.impl.BlockSupportImpl.reparse(BlockSupportImpl.java:101)
	at com.intellij.psi.impl.DocumentCommitThread.doCommit(DocumentCommitThread.java:682)
	at com.intellij.psi.impl.DocumentCommitThread.lambda$commitUnderProgress$4(DocumentCommitThread.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1129)
	at com.intellij.psi.impl.DocumentCommitThread.commitUnderProgress(DocumentCommitThread.java:473)
	at com.intellij.psi.impl.DocumentCommitThread.commitSynchronously(DocumentCommitThread.java:364)
	at com.intellij.psi.impl.PsiDocumentManagerBase.lambda$doCommit$1(PsiDocumentManagerBase.java:447)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.psi.impl.PsiDocumentManagerBase.doCommit(PsiDocumentManagerBase.java:455)
	at com.intellij.psi.impl.PsiDocumentManagerBase.commitDocument(PsiDocumentManagerBase.java:295)
	at com.intellij.codeInsight.editorActions.enter.EnterBetweenBracesDelegate.bracesAreInTheSameElement(EnterBetweenBracesDelegate.java:22)
	at com.intellij.codeInsight.editorActions.enter.EnterBetweenBracesFinalHandler.isApplicable(EnterBetweenBracesFinalHandler.java:79)
	at com.intellij.codeInsight.editorActions.enter.EnterBetweenBracesFinalHandler.preprocessEnter(EnterBetweenBracesFinalHandler.java:42)
	at com.intellij.codeInsight.editorActions.EnterHandler.a(EnterHandler.java:128)
	at com.intellij.codeInsight.editorActions.EnterHandler.b(EnterHandler.java:81)
	at com.intellij.psi.impl.source.PostprocessReformattingAspect.lambda$disablePostprocessFormattingInside$1(PostprocessReformattingAspect.java:110)
	at com.intellij.psi.impl.source.PostprocessReformattingAspect.disablePostprocessFormattingInside(PostprocessReformattingAspect.java:118)
	at com.intellij.psi.impl.source.PostprocessReformattingAspect.disablePostprocessFormattingInside(PostprocessReformattingAspect.java:109)
	at com.intellij.codeInsight.editorActions.EnterHandler.executeWriteAction(EnterHandler.java:80)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler$1.run(EditorWriteActionHandler.java:51)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler.doExecute(EditorWriteActionHandler.java:64)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.execute(EditorActionHandler.java:210)
	at com.intellij.codeInsight.template.impl.editorActions.EnterHandler.executeWriteAction(EnterHandler.java:49)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler$1.run(EditorWriteActionHandler.java:51)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.openapi.editor.actionSystem.EditorWriteActionHandler.doExecute(EditorWriteActionHandler.java:64)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.lambda$null$2(EditorActionHandler.java:202)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.doIfEnabled(EditorActionHandler.java:110)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.lambda$execute$3(EditorActionHandler.java:201)
	at com.intellij.openapi.editor.impl.CaretModelImpl.a(CaretModelImpl.java:363)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:452)
	at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:355)
	at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:346)
	at com.intellij.openapi.editor.actionSystem.EditorActionHandler.execute(EditorActionHandler.java:201)
	at com.intellij.openapi.editor.actionSystem.EditorAction.lambda$actionPerformed$0(EditorAction.java:98)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.a(CoreCommandProcessor.java:137)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:95)
	at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:107)
	at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:86)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:220)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:237)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:578)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.a(IdeKeyEventDispatcher.java:627)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:626)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.a(IdeKeyEventDispatcher.java:478)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:212)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:702)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:395)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

The last thing I did was writing:
```reason
if (something) {[ENTER]}
```
and hit new line where it says `[ENTER]`. 

It started working again after a restart.

**Comment by giraud** [comment]
Working on it, found the root cause. I hope to release a new version today/tomorrow

**Comment by giraud** [comment]
fixed in v0.58.1
